coin/coin_test.go

```diff
@@ -7,148 +7,158 @@ import (
 	"testing"
 
 	"github.com/iov-one/weave/errors"
-	"github.com/stretchr/testify/assert"
-)
-
-type cmp int
-
-const (
-	neg  cmp = -1
-	zero     = 0
-	pos      = 1
+	"github.com/iov-one/weave/weavetest/assert"
 )
 
 func TestCompareCoin(t *testing.T) {
-
-	cases := []struct {
-		a      Coin
-		b      Coin
-		expect int
-		aState cmp
-		bState cmp
+	cases := map[string]struct {
+		a       Coin
+		b       Coin
+		wantRes int
 	}{
-		{
-			NewCoin(20, 1234, "ABC"),
-			NewCoin(19, 999999999, "ABC"),
-			1,
-			pos,
-			pos,
-		},
-		{
-			NewCoin(0, -2, "FOO"),
-			NewCoin(0, 1, "FOO"),
-			-1,
-			neg,
-			pos,
-		},
-		{
-			NewCoin(-4, -2456, "BAR"),
-			NewCoin(-4, -4567, "BAR"),
-			1,
-			neg,
-			neg,
-		},
-		{
-			Coin{},
-			Coin{},
-			0,
-			zero,
-			zero,
+		"a greater than b": {
+			a:       NewCoin(20, 1234, "ABC"),
+			b:       NewCoin(19, 999999999, "ABC"),
+			wantRes: 1,
+		},
+		"a smaller than b": {
+			a:       NewCoin(0, -2, "FOO"),
+			b:       NewCoin(0, 1, "FOO"),
+			wantRes: -1,
+		},
+		"a greater than b and both negative": {
+			a:       NewCoin(-4, -2456, "BAR"),
+			b:       NewCoin(-4, -4567, "BAR"),
+			wantRes: 1,
+		},
+		"zero value coins": {
+			a:       Coin{},
+			b:       Coin{},
+			wantRes: 0,
 		},
 	}
 
-	for idx, tc := range cases {
-		t.Run(fmt.Sprintf("case-%d", idx), func(t *testing.T) {
-			// make sure both show proper results
-			assert.Equal(t, tc.a.IsZero(), tc.aState == zero)
-			assert.Equal(t, tc.a.IsPositive(), tc.aState == pos)
-			assert.Equal(t, !tc.a.IsNonNegative(), tc.aState == neg)
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			res := tc.a.Compare(tc.b)
```

Comment:
Why remove the IsZero/IsPositive/IsNonNegative tests?
You could move them into another test case (this one is doing two checks at once),
but it is good to ensure those cases are covered (maybe nicer than here)

Reply:
I only see `TestCoinIsPositive` below

Reply:
I will add separate test functions to test them.